client/views/edit/edit.js

```diff
@@ -0,0 +1,71 @@
+import { Instances } from '/lib/common.js';
+
+function showEditAdvError(reason) {
+  if (typeof currentError !== 'undefined') {
+    Blaze.remove(currentError);
+  }
+  const parentNode = document.getElementsByClassName('formcontainer')[0];
+  const nextNode =  document.getElementsByClassName('inputcontainer')[0];
+  currentError = Blaze.renderWithData(Template.form_error, reason, parentNode, nextNode);
+}
+
+Template.edit.onRendered(() => {
+  // When the template is rendered, set the document title
+  $('.formcontainer').hide().fadeIn(400);
+  $('#darker').hide().fadeIn(400);
+});
+
+Template.edit.helpers({
+  isSelected(property, value) {
+    return Template.instance().data.table[property] === value ? 'selected' : '';
+  },
+});
+
+Template.edit.events({
+  'click .editsubmitbutton': function (event, template) {
+    const table = Template.instance().data.table;
+    const threshold = template.$('select[name=threshold]').val();
+    const maxQuestion = template.$('select[name=max_question]').val();
+    const maxResponse = template.$('select[name=max_response]').val();
+    const redLength = template.$('select[name=new_length]').val();
+    const stale = template.$('select[name=stale_length]').val();
+    const isHidden = (template.$('select[name=visibility]').val() === 'hidden');
+    const social = (template.$('select[name=social]').val() === 'on');
+    const newValues = {
+      threshold: Number(threshold),
```

Comment:
Do you like `Number` more than `parseInt`?

Reply:
Here we are using Number() only to convert a string with numbers to a number and Number() returns NaN if gets character in the string unlike parseInt(). So shouldn't Number() be a better choice for us?

Reply:
I found the difference [here](https://stackoverflow.com/questions/4090518/what-is-the-difference-between-parseint-and-number) which is similar. But why do we want to allow `54af` since the input is a dropdown and `54af` clearly tells that there is a bug? Hence returning `NaN` for `54af` should be better in my opinion. Am I missing something here??

Reply:
We're not guaranteed the input will always be a dropdown. It's possible in the future someone changes it to a text box or something, in which case we might see inputs like this, and in that case, being permissive in what we accept seems like a good thing. This is super minor and shouldn't hold up the merge - I just generally subscribe to the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle). If you disagree, that's OK too.

Reply:
Yes, I agree with you. Will add this change too.